source/parser.hera

```diff
@@ -3871,14 +3871,15 @@ FunctionType
   Parameters __ "=>" Type
 
 TypeArguments
-  __ "<" __ Type (__ Comma __ Type)* (__ Comma)? __ ">"
+  __ "<" __ Type (__ Comma __ Type)* (__ Comma)? __ ">" ->
+    return { ts: true, children: $0 }
 
 TypeParameters
-  __ "<" __ Type (__ Comma __ Type)* (__ Comma)? __ ">" ->
+  __ "<" __ TypeParameter (__ Comma __ TypeParameter)* (__ Comma)? __ ">" ->
     return { ts: true, children: $0 }
 
 TypeParameter
-  __ Identifier TypeConstraint? TypeParameterDelimiter
+  __ Identifier TypeConstraint? # TypeParameterDelimiter
```

Comment:
@STRd6 I am not quite sure what was the purpose of the last part ...

Reply:
As in ts doesn't seem to care about trailing commas, so do we have a need to ?

Reply:
It's so we can do:

```
TypeParameters
  __ "<" TypeParameter* __ ">" ->
    return { ts: true, children: $0 }
```

It looks like this was partially implemented, just not hooked up.

TypeParameterDelimiter could also be simplified by removing the `&( _* ">" )` assertion and changing `&( __ ">" )` to not insert a comma.

Reply:
Ok, updated. The `  &EOS InsertComma -> $2` doesn't seem to be working for me though and I can't quite figure why

Reply:
I took a look and source/parser.hera:345 should be `TypeParameters` instead of `TypeArguments`.

Looks great overall!

Reply:
facepalm, updated.